### DISCUSSION TITLE
I've deleted all repos and will abandon this platform over 2FA requirement.

### DISCUSSION THREAD

**Discussion by ghost** [community/community#63846] [Discussions] [answered]
### Select Topic Area

Platform Abandonment

### Body

I am a hobby programmer, I simply hosted a few game projects on here.
I don't care at all about my profile security.
I've deleted all my repos and will abandon the platform.
I understand some larger profiles might want the feature for security, that should be an **opt-in** feature. Not forced upon us.
Peace out GitHub it was cool while it lasted.
Perhaps if they made the security feature a **choice** I might return one day, but as Micro$oft is at the helm that is unlikely to happen.

**Comment by ghost** [comment]
> That's unfortunate
You should try mailing them, they do answer to mails, I hope if you mail them your suggestion then they'll someday make it optional.

>> If you are satisfied with this answer, consider marking it as accepted, it means alot :-)

**Have a good day ahead.**

**Comment by msharov** [comment]
I also deleted all my projects and moved to SourceForge. Sadly, our protests will remain unnoticed, since we, as non-paying customers, don't matter. Heck, github will even count me as a successful convert, because I need to keep the account to report bugs in other projects, and so had to enable 2FA. Open source is dying. This is just another nail in its coffin.

**Reply by hexaheximal** [reply]
SourceForge is pretty dead - you should be moving to Codeberg or SourceHut instead.

Personally, I've chosen to migrate to Codeberg, and I'm really enjoying it.

**Reply by msharov** [reply]
Codeberg does not have webhosting, needed to host project documentation. SourceHut is a paid-only site, and since I make no money from my projects, paying for hosting is out of the question. Publishing my projects does not benefit me in any way, it's just for the users.

**Reply by hexaheximal** [reply]
> Codeberg does not have webhosting

That's not true - see: https://codeberg.page/

> SourceHut is a paid-only site

Fair enough. You can still self-host it if you have existing infrastructure though.

**Reply by msharov** [reply]
Well, Codeberg did not have webhosting when I moved off GitHub last year. Moving the projects is a pretty big hassle, and SourceForge works acceptably well, so I'm staying there.

As for self-hosting, that's also out of the question. There is no way to do it for free and I'm not making money from the projects.

**Reply by corwinn** [reply]
> Open source is dying.

No. Places like "github" are a direct proof that your statement is wrong. Do you really believe that they're spending exorbitant amount of resources to just shut it down? Yes their actions might be unpleasant - but we knew that from the moment we joined their game. Also, it doesn't matter whenever one is a paying customer or not - they do ignore me even thought I'm a paying customer (of their OS :) - not by choice); no point of further stating the obvious.

Anyways - the art* will never die; the ones with the surreal intelligence will make sure of it; - call me a dreamer if you must.

Just my 2c

**Comment by bronglil** [comment]
I understand your perspective on GitHub's security features and your preference for choice. While it's true that individual preferences vary, mandatory security measures often help protect the platform's integrity. If you ever reconsider, keep an eye on any updates that might address your concerns. Your feedback could contribute to positive changes in the future.

**Reply by msharov** [reply]
No, they don't. Enabling 2FA does not in any way help safeguard my account. I'm already using a strong, unique password, so there is no possibility whatsoever of it being compromised. 2FA is nothing but a nuisance to me. And even if it wasn't, it isn't GitHub's place to dictate to me what level of security is acceptable. All it has accomplished is to force me and my projects off the platform. I expect many experienced programmers will feel the same way I do, leaving GitHub with a pile of securely managed junk. I'll leave them to it.

**Reply by hexaheximal** [reply]
Exactly! I'm in the same scenario.

**Reply by adydeeram** [reply]
@bronglil Absolutely, your perspective on GitHub's security features and the importance of having choices is completely understood. Diversity in preferences is a natural part of any platform. At the same time, incorporating mandatory security measures plays a crucial role in safeguarding the overall integrity of the platform. It's worth noting that technology and platforms evolve, and updates are frequently introduced to enhance the user experience and address concerns. If you find the opportunity to revisit your stance, staying informed about these updates could be valuable. Your input and feedback have the potential to drive positive enhancements down the road.

**Reply by bronglil** [reply]
@msharov I understand your frustration with GitHub's mandatory 2FA. While security is vital, I believe it should be a personal choice. Forcing security measures can alienate experienced programmers like us. Flexibility could retain diverse users without compromising security. It's worth noting that GitHub does offer a range of security features to protect accounts and projects, such as encrypted storage, vulnerability scanning, and secure connections.

**Reply by msharov** [reply]
> It's worth noting that GitHub does offer a range of security features
> to protect accounts and projects, such as encrypted storage,
> vulnerability scanning, and secure connections.

All of which are only useful for web developers. All my projects are desktop Linux apps and daemons. I don't need online encrypted storage or secure connections. And, I write in C++. GitHub does not think it needs to support C++ for vulnerability scanning, dependency management, or anything else. That language must be just a passing fad, and if they ignore it long enough, it will surely go away as I did.

**Reply by bronglil** [reply]
GitHub's features and offerings continue to evolve over time based on user feedback and industry trends. While C++ might not have been a primary focus, it's possible that the platform could expand its support in the future, especially as more developers express their needs and preferences

stay happy with you perspective

**Comment by Jaber-Saed** [comment]
It sounds like you've decided to leave GitHub and are expressing your thoughts about their security features and Microsoft's involvement. While everyone has their own preferences and reasons for using or not using certain platforms, I'd like to provide some insights that might help you understand the perspective behind the security features on GitHub.

    Security Concerns: Security is a critical aspect of any online platform, especially one that deals with code and projects. GitHub hosts a vast amount of open-source and private repositories that could contain sensitive information. Introducing security features helps protect users' code and data from potential breaches or unauthorized access.

    Opt-In vs. Default: While you might not see the immediate need for security, many users, particularly those with larger projects, businesses, or projects with proprietary code, value these security measures. Making such features opt-in could potentially leave a large portion of the user base exposed to risks they might not fully understand.

    Microsoft's Involvement: Microsoft's acquisition of GitHub brought financial stability and resources to the platform. Microsoft has been working on integrating developer tools and services to enhance the overall experience. While opinions about Microsoft can vary, they have shown commitment to maintaining GitHub as a developer-friendly platform.

    Platform Evolution: Platforms evolve over time, and features may change to adapt to new demands, security concerns, or user needs. What seems unlikely today might become a possibility in the future.

    Choice and Flexibility: GitHub aims to provide a balance between features that cater to a wide range of users. It's important for platforms to consider both user preferences and security requirements when introducing changes.

If you decide to revisit GitHub in the future, you might find that their offerings have evolved to better suit your needs. Remember that technology landscapes are constantly changing, and reevaluation can sometimes lead to discovering solutions you hadn't considered before.

**Reply by hexaheximal** [reply]
There is an endless stream of reasons why you should abandon GitHub, including, but not at all limited to:

- https://nogithub.codeberg.page/
- https://librecmc.org/github.html
- https://sfconservancy.org/GiveUpGitHub/

Personally, I have moved to Codeberg: https://codeberg.org/hexaheximal

**Reply by msharov** [reply]
It sounds like you are using an AI comment generator in an attempt to make excuses for the company's bad decisions. You are failing miserably, and your "insights" are mostly nonsense:

> Security Concerns: Security is a critical aspect of any online platform

Absolutely. But this aspect is on GitHub's end, not mine. GitHub should indeed take measures to ensure my account is not compromised, but *on its end*, such as securely storing passwords so they can not be stolen in case of a website breach. It has no business dictating me what security measures *I* want to take on my end in safeguarding my login information.

> GitHub hosts a vast amount of open-source and private repositories
> that could contain sensitive information.

That is none of GitHub's business. If a repository contains sensitive information, server credentials, or whatever, it is the repository owner who is at fault here.

> Opt-In vs. Default: While you might not see the immediate need for security,
> many users, particularly those with larger projects, businesses, or projects
> with proprietary code, value these security measures.

Indeed they do. Businesses want to leach off other people's work in open source without paying for it and then they complain that some of this work contains security holes or gets compromised. Well, I have an alternative suggestion: if these leeches want to have security guarantees for my projects, they should pay me for it. Doing 2FA is a nuisance and pointless work, but I'm game if you want to pay me to enable it. Say, $10,000/year will be adequate.

> Making such features opt-in could potentially leave a large portion
> of the user base exposed to risks they might not fully understand.

I'll argue that they *should* be exposed to any risk they do not fully understand. I'll be happy to explain any risks associated with their use of my projects, for a reasonable fee.

> While opinions about Microsoft can vary, they have shown commitment
> to maintaining GitHub as a developer-friendly platform.

On the contrary, since Microsoft's acquisition, GitHub has become an increasingly hostile environment for anyone who is not a corporate web developer. Even without the 2FA mandate, I was likely going to leave just because of all the new "features".

> It's important for platforms to consider both user preferences
> and security requirements when introducing changes.

Indeed. And GitHub clearly is not interested in doing either one. Is it any wonder that people are leaving?

**Comment by msharov** [comment, marked as answer]
I would also like to address GitHub's constant harping on how 2FA improves security, when in reality it does not. While GitHub does indeed benefit from being able to say that all its accounts are now unhijackable, the change for users is much for the worse. I've focused mainly on the aspect of forced unpaid labor (aka slavery) that 2FA imposes on us all, and the visceral reaction many of us have to this imposition. But are our accounts actually more secure because of all this extra work? Quite the contrary.

First, the obsession with account hijacking in the first place is misplaced. There is only a very small number of repositories that are targeted for this, and I'm talking miniscule here - maybe a few hundred on all of GitHub. Nobody is going to even try to hijack my account, since my projects are not part of any critical supply chain. Focusing only on the vulnerable accounts would have been a far better choice.

Second, account hijacking is only a problem because of the idiotic design of NPM, which pulls code directly from the repository without any kind of verification of what's in it. That is the main reason a hijacker is able to introduce a vulnerability into the code. On any of my own repositories, for example, any such attempt will fail because I am the sole developer and I will immediately detect any unauthorized changes the next time I try to push my own. Furthermore, nobody pulls my repositories directly. Unlike in the insane world of Javascript, a C++ application makes releases and it is only those releases that are used by downstream packagers. I sign all the releases; there is no way anybody can introduce anything unauthorized there.

To fix the security problem here, 2FA is not the answer; you need to get rid of NPM and all these numerous automatic dependencies. Each project *must* vet the code of its dependencies and *must* manually verify each new release before trusting it. This is not just a question of the account being hijacked. You must realize that identification is only one factor in establishing trust and each identified account holder is not necessarily trustworthy. Criminals have identification too, you know, and you can't discover criminal intent by just verifying the author's identity. You must verify the code itself, or, at the very least, the author's reputation. Yes, it can be a lot of work, and that's a good thing because it encourages you to minimize the number of dependencies. Don't include random repositories. Use only well-known, mature libraries.

Which brings us to reputation and how 2FA harms it by making account recovery significantly more difficult or even impossible. For most of us, losing control of our email account is an event far more rare and unlikely than losing the password or the rather voluminous recovery codes. The traditional reset-by-email account recovery works well for everyone and provides adequate amount of security. With 2FA and something going wrong with it, as it always does with these horrendously designed third party TOTP authenticator apps, losing the recovery codes means losing the account. Microsoft is a big corp that provides no customer support. If you can't get the account recovered by the normal automated process, you are out of luck. The corp is not going to listen to your reasons or make any exceptions for your situation. If you lose your codes, you're totally screwed.

Losing the account means that I would now have my code publicly available on GitHub, but having no access to it. That code may become obsolete, or a vulnerability may be discovered in it, and I would be unable to do a damn thing about it. This directly damages my reputation, which is the only assurance you get about the quality of my code. By increasing the likelihood of account lockout, GitHub is thus damaging the security of the entire supply chain.

This is an even greater problem for Javascript developers, who are GitHub's main target audience. My projects are distributed through releases. The actual location of the repository is of less importance for me, since if locked out, I would merely have to notify the distribution packagers of the new location. This is a bit of work, but no worse than what I already had to do when I moved my projects to SourceForge. For NPM, the repo is the release. Lose the repo, and you're totally screwed. You lose your code, you lose your reputation, and you lose any possibility that anything you ever write again will be used by anybody.

This is what makes 2FA such a rotten deal that you should avoid whenever possible. To repeat: it is much, much more likely that you'll lose your 2FA and your recovery codes than it is that you would lose your email account. 2FA makes it much more likely that you will lose your account, and makes this a much more likely possibility than getting your account hijacked, the chance of which is miniscule for the vast majority of users. Losing your account is also likely to be much more damajing the the hijacking. And knowing this, the next time you hear some GitHub drone tell you how much more "secure" he is making you, slap him upside the head.

**Comment by kuroodo** [comment]
I tried speaking to Github support over this and they won't budge. They also keep throwing out these technical terms they made up and use to justify their decision. I can understand them wanting this for major organizations and big time users. But for the majority of people in this platform, it's no good.

I've migrated to GitLab and recommend others do too. Cancelled my copilot subscription as well despite me loving the tool so much. Shame

**Comment by Fish-Git** [comment]
> > Security Concerns: Security is a critical aspect of any online platform
> 
> Absolutely. But this aspect is on GitHub's end, not mine. GitHub should indeed take measures to ensure my account is not compromised, but _on its end_, such as securely storing passwords so they can not be stolen in case of a website breach. It has no business dictating me what security measures _I_ want to take on my end in safeguarding my login information.

_**PRECISELY!**_ This 2FA crap is not designed to protect users _(regardless of how long and how often they continue to promote that false claim)_. It is designed to protect _them_.

**Comment by kenmcd** [comment]
> _**PRECISELY!**_ This 2FA crap is not designed to protect users _(regardless of how long and how often they continue to promote that false claim)_. It is designed to protect _them_.

And for them to be able tie your account(s) with your phone number.
Makes it easy to connect you to every other thing you do on the net.

Constantly having to block phone apps on Windows now (trying to connect to MS).
Every time it updates it has a new path which breaks the firewall block.
Of course MS would never try to connect everyone here with their phones, their PCs, and all other online activity.
